ext/mbstring/libmbfl/mbfl/mbfilter.c

```diff
@@ -1203,31 +1203,35 @@ mbfl_strcut(
 	return result;
 }
 
-
-/*
- *  strwidth
- */
-static size_t is_fullwidth(int c)
+/* Some East Asian characters, when printed at a terminal (or the like), require double
+ * the usual amount of horizontal space. We call these "fullwidth" characters. */
+static size_t character_width(int c)
 {
-	int i;
-
-	if (c < mbfl_eaw_table[0].begin) {
-		return 0;
-	}
-
-	for (i = 0; i < sizeof(mbfl_eaw_table) / sizeof(mbfl_eaw_table[0]); i++) {
-		if (mbfl_eaw_table[i].begin <= c && c <= mbfl_eaw_table[i].end) {
-			return 1;
+	/* The first fullwidth codepoint is 0x1100; the most commonly used characters,
+	 * including all those in the ASCII character set, are below that */
+	if (c < 0x1100) {
```

Comment:
Check against `mbfl_eaw_table[0][0]`?

Reply:
Good idea. That will be more future-proof, in case the data in the tables ever changes.